internal/manager/manager.go

```diff
@@ -198,3 +277,22 @@ func getProcessedInputs(ctx context.Context, repo MachineRepository, appAddress
 	f := repository.InputFilter{NotStatus: Pointer(InputCompletionStatus_None)}
 	return repo.ListInputs(ctx, appAddress, f, repository.Pagination{})
 }
+
+// Helper function to get inputs after a specific index
+func getInputsAfterSnapshot(ctx context.Context, repo MachineRepository, app *Application, snapshotInputIndex uint64) ([]*Input, error) {
```

Comment:
You could also take the slice after finding the index. Probably doesn't make much difference.

```
inputs, _, err := getProcessedInputs(ctx, repo, app.IApplicationAddress.String())
if err != nil {
	return nil, err
}

for i, input := range inputs {
	if input.Index > snapshotInputIndex {
		return inputs[i:], nil
	}
}
return []*Input{}, nil
```